libs/core/src/lib/popover/popover-position/popover-position.ts

```diff
@@ -0,0 +1,103 @@
+    'left': { originX: 'start', originY: 'center', overlayX: 'end', overlayY: 'center' },
+    'left-end': { originX: 'start', originY: 'bottom', overlayX: 'end', overlayY: 'bottom' },
+    'right-start': { originX: 'end', originY: 'top', overlayX: 'start', overlayY: 'top' },
+    'right': { originX: 'end', originY: 'center', overlayX: 'start', overlayY: 'center' },
+    'right-end': { originX: 'end', originY: 'bottom', overlayX: 'start', overlayY: 'bottom' }
+};
+
+export const DefaultPositions: ConnectedPosition[] = [
+    { originX: 'start', originY: 'bottom', overlayX: 'start', overlayY: 'top' },
+    { originX: 'start', originY: 'top', overlayX: 'start', overlayY: 'bottom' },
+    { originX: 'center', originY: 'bottom', overlayX: 'center', overlayY: 'top' },
+    { originX: 'center', originY: 'top', overlayX: 'center', overlayY: 'bottom' },
+    { originX: 'end', originY: 'bottom', overlayX: 'end', overlayY: 'top' },
+    { originX: 'end', originY: 'top', overlayX: 'end', overlayY: 'bottom' }
+];
+
+export type PopoverFillMode = 'at-least' | 'equal';
+
+const ARROW_SIZE  = '0.5rem';
+
+
+export type Placement = 'auto-start'
+    | 'auto'
+    | 'auto-end'
+    | 'top-start'
+    | 'top'
+    | 'top-end'
+    | 'right-start'
+    | 'right'
+    | 'right-end'
+    | 'bottom-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'left-end'
+    | 'left'
+    | 'left-start';
+
+export type XPositions = 'start' | 'center' | 'end';
+export type YPositions = 'top' | 'center' | 'bottom';
+export type ArrowPosition = 'top' | 'bottom' | 'start' | 'end';
+
+export class PopoverPosition {
+
+    static getCdkPlacement(placement: Placement): ConnectedPosition {
+        const resultCdkPlacement = popoverPlacementMap[placement];
+
+        if (!resultCdkPlacement && isDevMode()) {
+            throw new Error('Invalid function argument. Check if "placement" is type of Placement Union');
+        }
+
+        return resultCdkPlacement;
+    }
+
+    static getArrowPosition(position: ConnectedPosition, rtl?: boolean): ArrowPosition {
+        let _position: ArrowPosition = null;
+
+        if (position.overlayY !== position.originY &&
+            position.originY !== 'center' &&
+            position.overlayY !== 'center') {
+            _position = position.overlayY;
+        } else if (
+            position.overlayX !== position.originX &&
+            position.overlayX !== 'center' &&
+            position.originX !== 'center') {
+            _position = position.overlayX;
+
+            if (rtl) {
+                if (_position === 'start') {
+                    _position = 'end';
+                } else if (_position === 'end') {
+                    _position = 'start';
+                }
+            }
+        }
+
+        return _position;
+    }
+
+    static getMarginStyle(position: ArrowPosition): string {
+        if (!position) {
+            return;
+        }
+        if (position === 'start') {
+            return 'margin-left: 0.5rem;';
```

Comment:
Does not respects `ARROW_SIZE` for `start` and `end` position.

Reply:
It has been improved. Thanks for catching it